src/Compiler.php

```diff
@@ -114,18 +121,28 @@ private function getArrayProperty()
         $token = array_shift($this->request);
 
         if (!self::scanProperty($token, $array)) {
-            return false;
+            throw new Exception(
+                self::ERROR_NO_INITIAL_PROPERTY,
+                array('token' => $token),
+                self::ERROR_NO_INITIAL_PROPERTY .
+                " Error: API requests must begin with a property."
+            );
         }
 
         list($class, $path) = $this->schema->getPropertyDefinition('Database', $array);
-        $list = array_pop($path);
 
-        if (!isset($class, $path, $list)) {
+        if (!isset($class, $path)) {
             return false;
         }
 
+        $list = array_pop($path);
+
         list($table, $id, $hasZero) = $this->schema->getListDefinition($list);
```

Comment:
We should test this in the vagrant environment: Will this statement ("list($table, $id, $hasZero) = null;") generate a warning in PHP 5.3? It won't in PHP 5.5. But it might in the version of PHP that is running on the web server.